src/main/java/com/pinterest/secor/reader/MessageReader.java

```diff
@@ -65,7 +66,12 @@ public MessageReader(SecorConfig config, OffsetTracker offsetTracker) throws
 
         mConsumerConnector = Consumer.createJavaConsumerConnector(createConsumerConfig());
 
-        TopicFilter topicFilter = new Whitelist(mConfig.getKafkaTopicFilter());
+        if (!mConfig.getKafkaTopicBlacklist().isEmpty() && !mConfig.getKafkaTopicFilter().isEmpty()) {
+            throw new RuntimeException("Topic filter and blacklist cannot be both specified.");
+        }
+        TopicFilter topicFilter = !mConfig.getKafkaTopicBlacklist().isEmpty()? new Blacklist(mConfig.getKafkaTopicBlacklist()):
```

Comment:
If it is not specified instead of empty string, it will not pass https://github.com/pinterest/secor/blob/master/src/main/java/com/pinterest/secor/common/SecorConfig.java#L458.
I also tested both empty value, it runs but empty WhiteList.